### PR TITLE
Implement RobotFramework variants

### LINKED ISSUE #157: Implement RobotFramework variants
Acceptance criteria:
- enable a variant mechanism to use the same RobotFramework test (as requested by QE) for variants: { REST flow using Go proxy in `/cmd`, Python flow using client in `clients/python`}
- demonstrate can be used for User Stories
- demonstrate can be used to prove logical schema and logical mapping

### PR DESCRIPTION
**Pull request by tarilabs**
Resolves #157 

## Description
Demonstrates how to:
- enable a variant mechanism to use the same RobotFramework test (as requested by QE) for variants: { REST flow using Go proxy in `/cmd`, Python flow using client in `clients/python`}
- demonstrate can be used for User Stories
- demonstrate can be used to prove logical schema and logical mapping

Highlight semantic difference in data model used in REST (defined in OpenAPI) Vs Python client
for example in python client there is a required ModelArtifact and a required author:
https://github.com/opendatahub-io/model-registry/blob/9f3f9a219ded31470d674bacc4250f2f1e5a4923/clients/python/src/model_registry/types/contexts.py#L50-L52

(here we see 3 required properties)

which is not the case for OpenAPI generated model:
https://github.com/opendatahub-io/model-registry/blob/9f3f9a219ded31470d674bacc4250f2f1e5a4923/pkg/openapi/model_model_version.go#L21-L35

(here we see none required properties)

As coming from the OpenAPI contract:

https://github.com/opendatahub-io/model-registry/blob/9f3f9a219ded31470d674bacc4250f2f1e5a4923/api/openapi/model-registry.yaml#L1041-L1054

## How Has This Been Tested?
```sh
# start MLMD
docker run -p 9090:8080 --env METADATA_STORE_SERVER_CONFIG_FILE=/tmp/shared/conn_config.pb --volume ./test/bdd:/tmp/shared gcr.io/tfx-oss-public/ml_metadata_store_server:1.14.0

# start ModelRegistry
./model-registry proxy  

TEST_MR_HOST=host.docker.internal robot test/robot/UserStory.robot
TEST_MR_HOST=host.docker.internal robot test/robot/MRandLogicalModel.robot
TEST_MR_HOST=host.docker.internal TEST_MODE=Python robot test/robot/MRandLogicalModel.robot
```

You can omit `TEST_MR_HOST=host.docker.internal` if not used in emulated/DevContainer environment

## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: clients/python/src/model_registry/types/artifacts.py [hunk 1]
```diff
@@ -93,9 +93,14 @@ class ModelArtifact(BaseArtifact, Prefixable):
...
 
+    _model_version_id: Optional[str] = field(init=False, default=None)
+
     @property
     def mlmd_name_prefix(self) -> str:
-        return uuid4().hex
+        if self._model_version_id:
+            return self._model_version_id
+        else:
+            return uuid4().hex
```
Comment (tarilabs) [inline clients/python/src/model_registry/types/artifacts.py:103]: @isinyaaa kindly requesting specifically for your review also here, in addition to what described in the PR description (which is best be accounted indeed in a separate PR if required)

**Comment by lampajr** [comment]
lgtm!

But as I actively participated in the implementation, I would let final approval/review from another maintainer.

**Comment by tonyxrmdavidson** [comment]
lgtm from QE perspective

**Review by antisaling — APPROVED** [review]
lgtm
